lua/lspconfig/ltex.lua

```diff
@@ -0,0 +1,186 @@
+                client.config.settings.ltex.disabledRules = {
+                    [lang] = readFiles(client.config.disabledrules_files[lang])
+                };
+                return client.notify('workspace/didChangeConfiguration', client.config.settings)
+            else
+                return vim.notify("Error when reading disabledRules config, check it")
+            end
+
+        elseif configtype == 'falsePositive' then
+            if client.config.settings.ltex.hiddenFalsePositives then
+                client.config.settings.ltex.hiddenFalsePositives = {
+                    [lang] = readFiles(client.config.falsepositive_files[lang])
+                };
+                return client.notify('workspace/didChangeConfiguration', client.config.settings)
+            else
+                return vim.notify("Error when reading hiddenFalsePositives config, check it")
+            end
+        end
+    else
+        return nil
+    end
+end
+
+local function addToFile(filetype, lang, file, value)
+    file = io.open(file[#file-0], "a+") -- add only to last file defined.
+    if file then
+        file:write(value .. "\n")
+        file:close()
+    else
+        return print("Failed insert %q", value)
+    end
+    if filetype == 'dictionary' then
+        return updateConfig(lang, "dictionary")
+    elseif filetype == 'disable' then
+        return updateConfig(lang, "disable")
+    elseif filetype == 'falsePositive' then
+        return updateConfig(lang, "falsePositive")
+    end
+end
+
+local function addTo(filetype, lang, file, value)
+    local dict = readFiles(file)
+    for _, v in ipairs(dict) do
+        if v == value then
+            return nil
+        end
+    end
+    return addToFile(filetype, lang, file, value)
+end
+
+configs.ltex = {
+    default_config = {
+        cmd = {"ltex-ls"};
+        filetypes = {'tex', 'bib', 'md'};
+        root_dir = function(filename)
+            return util.path.dirname(filename)
+        end;
+        dictionary_files = { ["en"] = {vim.fn.getcwd() .. "dictionary.ltex"} };
+        disabledrules_files = { ["en"] = {vim.fn.getcwd() .. "disable.ltex"} };
+        falsepositive_files = { ["en"] = {vim.fn.getcwd() .. "false.ltex"}};
+        settings = {
+            ltex = {
+                enabled= {"latex", "tex", "bib", "md"},
+                checkFrequency="save",
+                language="en",
+                diagnosticSeverity="information",
+                setenceCacheSize=2000,
+                additionalRules = {
+                    enablePickyRules = true,
+                    motherTongue= "en",
+                };
+                dictionary = {};
+                disabledRules = {};
+                hiddenFalsePositives = {};
+            },
+        };
+        on_attach = function(client, bufnr)
+                    -- local lang = client.config.settings.ltex.language
+            for lang,_ in ipairs(client.config.dictionary_files) do       --
```

Comment:
settings are automatically sent on_attach, I don't thin you need to update this.

Reply:
This came from my use case, where I use two languages in the same document with different dict files.

Do you have a better suggestion how to do this with multiple languages?